### PR TITLE
Support php8

### PR DESCRIPTION
**Pull request by vasilvestre**
This is the minimal changes I had to do in order to being able to use PHP 8 for installation.

I've added PHP 8 support to CI too. It's fine ?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -32,13 +32,13 @@
...
-        "doctrine/mongodb-odm": "~1.0",
+        "doctrine/mongodb-odm": "^2.2.0",
         "doctrine/orm": "~2.2",
         "phing/phing": "~2.4",
         "php-mock/php-mock-phpunit": "~1.0 || ~2.0",
+        "phpstan/phpstan": "^0.12",
         "phpstan/phpstan-phpunit": "~0.9",
-        "phpstan/phpstan-shim": "~0.9",
-        "phpunit/phpunit": "~5.0 || ~6.0",
+        "phpunit/phpunit": "~9.0",
```
Comment (deguif) [inline composer.json:41]: This won't support PHP 7.2 as `phpunit/phpunit:^9.0` requires PHP 7.3 , but as CI is broken, I'm fine with it.
Need to spend some time to migrate CI to github action and make it work again.

**Review by deguif — APPROVED** [review]

**Comment by deguif** [comment]
@vasilvestre can you cancel your last commit.
Coding styles checks doesn't need to be run on multiple PHP versions.

File: composer.json (hunk 1 above)
Comment (vasilvestre) [reply composer.json:41]: I had to in order to use php 8 sadly :/ Github action work nicely and I got some experience. Could help, feel free to reach me :)

File: composer.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
         }
     ],
     "require": {
-        "php": "^7.2",
+        "php": "^8.0 || ^7.2",
```
Comment (deguif) [inline composer.json:23]:
```suggestion
        "php": "^7.2 || ^8.0",
```

File: composer.json [hunk 3]
```diff
@@ -32,13 +32,13 @@
     },
     "require-dev": {
         "doctrine/doctrine-bundle": "^2.0",
-        "doctrine/mongodb-odm": "~1.0",
+        "doctrine/mongodb-odm": "^2.2.0",
```
Comment (deguif) [inline composer.json:35]:
```suggestion
        "doctrine/mongodb-odm": "^2.2",
```

File: composer.json (hunk 1 above)
Comment (deguif) [inline composer.json:41]:
```suggestion
        "phpunit/phpunit": "^9.0",
```

**Review by deguif — APPROVED** [review]

**Pull request merged**